Replace std::random_shuffle with std::shuffle

tomsmeding opened the pull request:
According to [the docs][1], `std::random_shuffle` is deprecated in C++14 and removed in C++17. Since `std::shuffle` is available since C++11 as a replacement and monero already requires C++11, I think it is a good idea to use `std::shuffle` instead.

The choice of random number generator was inspired by the other usages of `std::shuffle` in the codebase; since the original `std::random_shuffle` made no guarantees whatsoever about the quality of its randomness, this does not introduce problems there.

[1]: https://en.cppreference.com/w/cpp/algorithm/random_shuffle

vtnerd commented on src/p2p/net_peerlist.h:
```diff
@@ -290,7 +290,7 @@ namespace nodetool
     if (anonymize)
     {
-      std::random_shuffle(bs_head.begin(), bs_head.end());
+      std::shuffle(bs_head.begin(), bs_head.end(), std::default_random_engine(crypto::rand<unsigned>()));
```
I would switch this to:
```c++
std::shuffle(bs_head.begin(), bs_head.end(), crypto::random_device{})
```
the `std::default_random_engine` doesn't have to be a "cryptographically secure" RNG. Otherwise this should be suitable for test purposes.

tomsmeding replied:
This would of course be an easy change. If this is necessary here, though, does this solve a bug in the original code? std::random_shuffle also doesn't use a cryptographically secure RNG.

vtnerd replied:
Yes, this was/is a bug in the existing code.

tomsmeding commented:
I addressed @vtnerd's comments (thanks!) in the latest commit; I replaced all instances of std::shuffle, not only the important ones, to prevent people from being able to copy an "insecure" call over to a new call site where a "secure" one would be warranted. In that sense, this is a form of defense-in-depth. I think performance doesn't suffer too much from this.

@moneromooo-monero

selsta commented:
Can you squash the commits together? :)

tomsmeding commented:
@selsta Of course, that makes a lot of sense here. :) Done just now.

tomsmeding commented:
@selsta Thanks for the quick turnaround!
